### PR TITLE
Further bug fixes to the charge version workflow process

### PR DESCRIPTION
**Pull request by timtamimi**
Explanation of what this PR is about: https://eaflood.atlassian.net/browse/WATER-3070?focusedCommentId=290138

### CODE REVIEW

File: src/internal/modules/charge-information/controllers/charge-information.js [hunk 1]
```diff
@@ -204,12 +204,16 @@ const updateDraftChargeInformation = async (request, h) => {
 
   const preparedChargeInfo = prepareChargeInformation(id, draftChargeInformation);
   preparedChargeInfo.chargeVersion['status'] = 'draft';
-
+  const { user_id: userId, user_name: userName } = get(request, 'defra.user');
```
Comment (Stephan-Hough) [inline src/internal/modules/charge-information/controllers/charge-information.js:207]: Adding the user details here will overwrite the details of the user who originally submitted the draft charge information.

File: src/internal/modules/charge-information/controllers/view-charge-information.js [hunk 2]
```diff
@@ -88,11 +88,16 @@ const postReviewChargeInformation = async (request, h) => {
     if (request.payload.reviewOutcome === 'approve') {
       await services.water.chargeVersions.postCreateFromWorkflow(request.params.chargeVersionWorkflowId);
     } else {
+      const { user_id: userId, user_name: userName } = get(request, 'defra.user');
+      const patchObject = {
+        status: request.payload.reviewOutcome,
+        approverComments: request.payload.reviewerComments,
+        chargeVersion: {},
+        createdBy: { id: userId, email: userName }
```
Comment (Stephan-Hough) [inline src/internal/modules/charge-information/controllers/view-charge-information.js:96]: similar to my previous comment - this was not the intended business logic. Not sure if this has changed

**Review by Stephan-Hough — CHANGES_REQUESTED** [review]
A couple of comments added regarding the pathing of the user details to the draft charge info and creating a new charge version

File: src/internal/modules/charge-information/controllers/view-charge-information.js (hunk 2 above)
Comment (timtamimi) [reply src/internal/modules/charge-information/controllers/view-charge-information.js]: You're right. This was a misunderstanding on my part. I have removed this.

File: src/internal/modules/charge-information/controllers/charge-information.js (hunk 1 above)
Comment (timtamimi) [reply src/internal/modules/charge-information/controllers/charge-information.js]: You're right. This was a misunderstanding on my part. I have removed this.

**Review by Stephan-Hough — APPROVED** [review]

**Pull request merged**